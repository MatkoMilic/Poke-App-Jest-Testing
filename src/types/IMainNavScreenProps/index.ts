export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
