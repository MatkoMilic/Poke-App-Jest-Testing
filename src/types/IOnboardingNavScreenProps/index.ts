export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
