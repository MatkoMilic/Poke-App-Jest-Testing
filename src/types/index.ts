export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
export type {
  MainNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
  RootNavigatorKeys,
  RootNavigatorParamList,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorType,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
export type {IUserValues} from './IUserValues/IUserValues';
export {initialUserLoginValues} from './IUserValues/IUserValues';
