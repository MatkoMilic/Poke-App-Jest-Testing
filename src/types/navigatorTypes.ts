import {NavigatorScreenParams} from '@react-navigation/native';

export enum OnboardingNavigatorScreens {
  LOADING_SCREEN = 'LoadingScreen',
  LOGIN_SCREEN = 'LoginScreen',
}
export type OnboardingNavigatorParamList = {
  LOADING_SCREEN: undefined;
  LOGIN_SCREEN: undefined;
};
export type OnboardingNavigatorKeys = keyof OnboardingNavigatorScreens;
export enum MainNavigatorScreens {
  PROFILE_SCREEN = 'ProfileScreen',
  POKELIST_SCREEN = 'PokeListScreen',
  SETTINGS_SCREEN = 'SettingsScreen',
}
export type MainNavigatorParamList = {
  PROFILE_SCREEN: undefined;
  POKELIST_SCREEN: undefined;
  SETTINGS_SCREEN: undefined;
};
export type MainNavigatorKeys = keyof MainNavigatorScreens;
export enum NavigatorNames {
  MainNavigator = 'MainNavigator',
  OnboardingNavigator = 'OnboardingNavigator',
}
export type NavigatorNameKeys = keyof NavigatorNames;
export type RootNavigatorType = {
  MainNavigator: NavigatorScreenParams<MainNavigatorParamList>;
  OnboardingNavigator: NavigatorScreenParams<OnboardingNavigatorParamList>;
};
export type RootNavigatorParamList = {
  MainNavigator: undefined;
  OnboardingNavigator: undefined;
};
export type RootNavigatorKeys = keyof RootNavigatorType;
