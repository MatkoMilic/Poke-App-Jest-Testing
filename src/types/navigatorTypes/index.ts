export type {
  MainNavigatorKeys,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorParamList,
  RootNavigatorType,
  RootNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
