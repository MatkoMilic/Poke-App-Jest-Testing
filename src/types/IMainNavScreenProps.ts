import {CompositeNavigationProp} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {RootNavigatorType, MainNavigatorParamList} from './navigatorTypes';

export type MainNavigationType = CompositeNavigationProp<
  NativeStackNavigationProp<MainNavigatorParamList>,
  NativeStackNavigationProp<RootNavigatorType>
>;
export interface IMainNavScreenProps {
  navigation: MainNavigationType;
}
