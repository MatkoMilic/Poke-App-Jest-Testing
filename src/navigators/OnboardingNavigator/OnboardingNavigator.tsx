import React, {FC} from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {LoadingScreen, LoginScreen} from '../../screens';
import {OnboardingNavigatorScreens} from '../../types';

const OnboardingStack = createNativeStackNavigator();

const OnboardingNavigator: FC = () => {
  return (
    <OnboardingStack.Navigator
      screenOptions={{headerShown: false}}
      initialRouteName={OnboardingNavigatorScreens.LOADING_SCREEN}>
      <OnboardingStack.Screen
        name={OnboardingNavigatorScreens.LOADING_SCREEN}
        component={LoadingScreen}
      />
      <OnboardingStack.Screen
        name={OnboardingNavigatorScreens.LOGIN_SCREEN}
        component={LoginScreen}
      />
    </OnboardingStack.Navigator>
  );
};

export default OnboardingNavigator;
