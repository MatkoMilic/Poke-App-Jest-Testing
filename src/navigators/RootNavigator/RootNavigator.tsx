import React, {FC} from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {OnboardingNavigator} from '../OnboardingNavigator';
import {MainNavigator} from '../MainNavigator';
import {NavigatorNames} from '../../types/navigatorTypes';

const RootStack = createNativeStackNavigator();
const RootNavigator: FC = () => {
  return (
    <RootStack.Navigator
      screenOptions={{headerShown: false}}
      initialRouteName={NavigatorNames.ONBOARDING_NAVIGATOR}>
      <RootStack.Screen
        name={NavigatorNames.ONBOARDING_NAVIGATOR}
        component={OnboardingNavigator}
      />
      <RootStack.Screen
        name={NavigatorNames.MAIN_NAVIGATOR}
        component={MainNavigator}
      />
    </RootStack.Navigator>
  );
};

export default RootNavigator;
