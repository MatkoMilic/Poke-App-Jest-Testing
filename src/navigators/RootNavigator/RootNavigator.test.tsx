import 'jsdom-global/register';
import React from 'react';
import {render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';
import RootNavigator from './RootNavigator';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  LoginScreen,
  PokeListScreen,
  ProfileScreen,
  SettingsScreen,
} from '../../screens';
import {OnboardingNavigator} from '../OnboardingNavigator';
import {MainNavigationType, OnboardingNavigationType} from '../../types';
configure({adapter: new Adapter()});

describe('Test many aspects of RootNavigator', () => {
  let mainNavigation: Partial<MainNavigationType>;
  let onboardingNavigation: Partial<OnboardingNavigationType>;
  beforeEach(() => {
    mainNavigation = {
      dispatch: jest.fn(),
    };
    onboardingNavigation = {
      dispatch: jest.fn(),
    };
  });
  const mountRootNavigator = mount(
    <NavigationContainer>
      <RootNavigator />
    </NavigationContainer>,
  );
  const renderRootNavigator = render(
    <NavigationContainer>
      <RootNavigator />
    </NavigationContainer>,
  );
  const isItOnboardingNavigator = mountRootNavigator.containsMatchingElement(
    <OnboardingNavigator />,
  );
  const testIfIsOnboardingNavigator = (isItOnboardingNavigator: boolean) =>
    isItOnboardingNavigator ? it : it.skip;

  test('check if RootNavigator renders', () => {
    expect(renderRootNavigator.toJSON()).toBeDefined();
  });
  it('check if RootNavigator mounts', () => {
    expect(mountRootNavigator.exists()).toBeTruthy();
  });
  testIfIsOnboardingNavigator(isItOnboardingNavigator)(
    'if we go to OnboardingNav check default screen',
    async () => {
      expect(mountRootNavigator.find('loadingText')).toBeDefined();
    },
  );
  it('check if another screen is also a child at the same mount', () => {
    expect(
      mountRootNavigator.containsMatchingElement(
        <ProfileScreen navigation={mainNavigation as MainNavigationType} />,
      ),
    ).toEqual(false);
    expect(mountRootNavigator.containsMatchingElement(<LoginScreen />)).toEqual(
      false,
    );
    expect(
      mountRootNavigator.containsMatchingElement(
        <PokeListScreen navigation={mainNavigation as MainNavigationType} />,
      ),
    ).toEqual(false);
    expect(
      mountRootNavigator.containsMatchingElement(
        <SettingsScreen navigation={mainNavigation as MainNavigationType} />,
      ),
    ).toEqual(false);
  });
  it('check is the proper navigator a child of RootNavigator', () => {
    expect(
      mountRootNavigator.containsMatchingElement(<OnboardingNavigator />),
    ).toEqual(true);
  });
  it('check if it matches the snapshot', () => {
    expect(mountRootNavigator).toMatchSnapshot();
  });
  it('check does unmount work on RootNavigator', () => {
    mountRootNavigator.unmount();
    expect(mountRootNavigator.exists()).toBeFalsy();
  });
});
