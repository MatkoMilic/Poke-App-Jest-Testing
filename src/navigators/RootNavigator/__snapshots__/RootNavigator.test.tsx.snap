// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test many aspects of RootNavigator check if it matches the snapshot 1`] = `ReactWrapper {}`;
